core/resource_manager.hpp

```diff
@@ -70,28 +74,31 @@ struct ResourceManagementOptions {
   IResourceManager* cached_columns{&IResourceManager::kNoop};
 };
 
+// TODO(MBkkt) rename to ManagedStdAllocator
 template<typename T>
 struct ManagedTypedAllocator
-  : ManagedAllocator<std::allocator<T>, IResourceManager> {
-  using Base = ManagedAllocator<std::allocator<T>, IResourceManager>;
+  : ManagedAllocator<IResourceManager, std::allocator<T>> {
+  using Base = ManagedAllocator<IResourceManager, std::allocator<T>>;
+
   explicit ManagedTypedAllocator()
-    : Base(
+    : Base{
 #ifdef IRESEARCH_DEBUG
         IResourceManager::kForbidden
 #else
         IResourceManager::kNoop
 #endif
-      ) {
+      } {
   }
+
   using Base::Base;
 };
 
 #ifdef __cpp_lib_polymorphic_allocator
 template<typename T>
-struct ManagedTypedPmrAllocator
-  : ManagedAllocator<std::pmr::polymorphic_allocator<T>, IResourceManager> {
-  using ManagedAllocator<std::pmr::polymorphic_allocator<T>,
-                         IResourceManager>::ManagedAllocator;
+struct ManagedPmrAllocator
+  : ManagedAllocator<IResourceManager, std::pmr::polymorphic_allocator<T>> {
+  using ManagedAllocator<IResourceManager,
+                         std::pmr::polymorphic_allocator<T>>::ManagedAllocator;
 };
```

Comment:
Such name more logical in my opinion:
Manager + Allocator -> ManagedAllocator<Managed, Allocator>
Managed + std::allocator -> ManagedStdAllocator
Managed + std::pmr::polymorphic_allocator -> ManagedPmrAllocator

Typed absolutely useless because any allocator should be typed, otherwise it's impossible to use (that reason why we are using struct instead of usings)